### DISCUSSION TITLE
Instagram limits

### DISCUSSION THREAD

**Discussion by thisisbobdole** [AAndyProgram/SCrawler#6] [Q&A] [answered]
Anyone know what the limits are for Instagram? I have everything set up finally now after some troubleshooting, and ripped an account that had 1400 posts, but it seemed to stop at a certain point (not sure when, 2200 items ripped due to lots of multiposts, but it appears to just go back a few years). I then tested on my own account and got the same thing, went back around 3.5 years then stopped. Re-running them gives an "all data downloaded" message at the bottom. I am also curious about account lockout limits, how many profiles/pictures can you download before you get banned?

Also, when I used ripmeapp it named the files based on the day they were posted, which was nice since it kept it in the same order. This appears to maybe just use the original filename? I am not sure how it is working exactly, the files are sort of in chronological order, but sometimes out of order a bit. Is there a way to add that functionality?

**Comment by AAndyProgram** [comment, marked as answer]
Please, read the [guide](https://github.com/AAndyProgram/SCrawler/wiki)

[Instagram limits](https://github.com/AAndyProgram/SCrawler/wiki/Settings#instagram-limits)
[Files naming](https://github.com/AAndyProgram/SCrawler/wiki/Settings#program-settings ): ```Change file names``` setting

I don't know ripmeapp!

> but it seemed to stop at a certain point (not sure when, 2200 items ripped due to lots of multiposts, but it appears to just go back a few years). I then tested on my own account and got the same thing, went back around 3.5 years then stopped

Have you seen the program log? I think you catch error 429 (too many requests). As I wrote in the [guide](https://github.com/AAndyProgram/SCrawler/wiki/Settings#instagram-limits), I'm still thinking about how to make Instagram algo smoother to avoid the 429 error.

I don't know how many posts a program can receive before the 429 error.

**Comment by AAndyProgram** [comment]
Hello. In the coming days I'll post a new release. This release will contain many improvements and an updated Instagram parsing algorithm among others things.

**About limits**
Now the algorithm receives 12 posts per request. 200 requests are can send to the server before a 429 error caught.
**In the next release, the number of received posts will be increased to 50 per request. Thus, the total number of posts will be increased from 2'400 to 10'000.** I also implemented several wait timers to get around this limitation and isolated the Instagram download jobs to a separated thread. This should help bypass the restrictions. I'm currently debugging a release and will post it soon.

**Comment by AAndyProgram** [comment]
@thisisbobdole please update to version 2.0.0.2
